### ISSUE TITLE
Support for custom path

### ISSUE THREAD

**Issue by JimmyBoh** [JimmyBoh/pully#4]
Currently the output path of a video is `./<author>/<title>.<extension>`. Marc-Antoine recommended a custom output path that can be specified in the options, like so:

```
var options = {
  url: 'http://www.youtube.com?v=',
  preset: 'audio'
  path: 'iwant/mydownload/here'
};
```

This should be a simple implementation when using `pully` programmatically, but via the command line it is a bit more complicated.

Since more options are imminent a better argument format might be necessary. The only breaking change might be specifying presets as `-p <preset>` instead of just `<preset>`.

Thoughts anyone?

**Comment by mariotacke** [comment]
I suggest a more verbose argument format than the current one. To keep it backwards compatible, we could simply add a fourth, optional, parameter such as:

`pully <url> [preset] [count] [output directory]`

For the new argument format I propose this:

- `-h`, `--help`, `/?`, as already implemented
- `-p preset`, `--preset preset`, `/preset preset`
- `-u url`, `--url url`, `/url url`
- `-c count`, `--count count`, `/count count`

The restructuring of the command line parameters probably deserves its own issue.

**Comment by JimmyBoh** [comment]
I agree @mariotacke, while the current format is simple, it doesn't lend well to more complex options. Most likely I will create Pully 2.0 with the new command line format, better support for format/encoding transformation, and possibly playlist/batch downloads. More on this to come!

**Comment by JimmyBoh** [comment]
In case anyone is wondering, I have started on 2.0, which is basically a re-write. This will be a good time to address the CLI argument structure (great ideas, @mariotacke) as well as update the project to use TypeScript, Promises, etc. Watch the [2.0-upgrade branch](https://github.com/JimmyBoh/pully/tree/2.0-upgrade) for progress.

**Comment by mariotacke** [comment]
Excellent. Glad you are jumping back on this! I wrote a small CLI tool a while back to do automated downloads on a schedule. If you need some inspiration, check it out [here](https://github.com/mariotacke/youtube-sync).

**Comment by JimmyBoh** [comment]
@mariotacke Thanks, I'll definitely take a look!

Something like that will be my ultimate goal, but in exactly what form is the question (simple console app, express server + mobile app for direct transfer to my phone, electron desktop app that just a player with download capabilities, all of the above). My thought is `pully` is only for downloading a video, and I just made [scany](https://github.com/JimmyBoh/scany) to easily scan user/playlist feeds, so hopefully with them combined we can make some pretty cool tools!

**Comment by JimmyBoh** [comment]
FYI this will soon be resolved after I implement the CLI feature of [Pully 2.0](https://github.com/JimmyBoh/pully/commits/2.0-upgrade)!

**Comment by JimmyBoh** [comment]
Implemented in 4c0937a. Feel free to re-open if any problems arise!
